StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -0,0 +1,212 @@
+                }
+                else if (!usingDirective.StaticKeyword.IsKind(SyntaxKind.None))
+                {
+                    staticImports.Add(usingDirective);
+                }
+                else
+                {
+                    var firstPart = usingDirective.Name;
+
+                    while (firstPart is QualifiedNameSyntax)
+                    {
+                        firstPart = ((QualifiedNameSyntax)firstPart).Left;
+                    }
+
+                    var firstPartName = ((IdentifierNameSyntax)firstPart).Identifier.ValueText;
+
+                    if (nameof(System).Equals(firstPartName, StringComparison.Ordinal))
+                    {
+                        systemUsings.Add(usingDirective);
+                    }
+                    else
+                    {
+                        otherUsings.Add(usingDirective);
+                    }
+                }
+            }
+
+            systemUsings.Sort(CompareName);
+            otherUsings.Sort(CompareName);
+            aliases.Sort(CompareAliasName);
+            staticImports.Sort(CompareName);
+
+            var rootNamespace = newSyntaxRoot.ChildNodes().OfType<NamespaceDeclarationSyntax>().FirstOrDefault();
+
+            var indentationLevel = IndentationHelper.GetIndentationSteps(indentationOptions, rootNamespace);
+            var usingsIndentation = IndentationHelper.GenerateIndentationString(indentationOptions, indentationLevel + 1);
+
+            var syntaxList = SyntaxFactory.List<UsingDirectiveSyntax>();
+            syntaxList = AppendUsings(syntaxList, systemUsings, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, otherUsings, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, aliases, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, staticImports, usingsIndentation);
+
+            if (rootNamespace != null)
+            {
+                var newRootNamespace = rootNamespace.WithUsings(syntaxList);
+                newSyntaxRoot = newSyntaxRoot.ReplaceNode(rootNamespace, newRootNamespace);
+            }
+            else
+            {
+                // There is no namespace, just add at the top of the file
+                newSyntaxRoot = ((CompilationUnitSyntax)newSyntaxRoot).WithUsings(syntaxList);
+            }
+
+            return newSyntaxRoot;
+        }
+
+        private static SyntaxList<UsingDirectiveSyntax> AppendUsings(SyntaxList<UsingDirectiveSyntax> syntaxList, List<UsingDirectiveSyntax> usings, string indentation)
+        {
+            if (usings.Count > 0)
+            {
+                var processedUsings = new List<UsingDirectiveSyntax>();
+
+                for (var i = 0; i < usings.Count; i++)
+                {
+                    var currentUsing = usings[i];
+                    var processedUsing = currentUsing.WithLeadingTrivia(SyntaxFactory.Whitespace(indentation)).WithTrailingTrivia(SyntaxFactory.CarriageReturnLineFeed);
```

Comment:
Then there would be inconsistency. There's lots of places where the same would happen, either with LF vs CRLF or space vs Tab. I'll create an issue to fix this somewhere in the future for the entire codebase.